### PR TITLE
Centralize app routes in global constant class

### PR DESCRIPTION
**Pull request by sven1103**

### CODE REVIEW

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java
Comment (Steffengreiner) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java:17]: I'd argue that MAIN, HOME or ROOT was actually correct here since it's intention is to be the route for directing the user to the current landing page.

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/overview/ProjectOverviewPage.java
Comment (Steffengreiner) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/overview/ProjectOverviewPage.java:24]: Would it make sense to also have additional aliases similiar to "home" or "main"? 
https://vaadin.com/docs/next/routing/route

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java
Comment (Steffengreiner) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java:11]: What is the reason for not defining the AppRoutes as an Enum?

**Review by Steffengreiner — CHANGES_REQUESTED** [review]
Heya @sven1103! Nicely done, i have some minor questions and will approve once they are answered :)

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java
Comment (sven1103) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java:11]: Why enums? we need the character sequence representation for the route annotation value property

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/overview/ProjectOverviewPage.java
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/project/overview/ProjectOverviewPage.java:24]: I understand but would argue that in the context of an application redirecting the user back to the main-page is a sensible operation. 
Without a "home" or "main" alias the knowledge that the ProjectOverviewPage functions as the landing page is implictit rather than explicit. 
However this is more of a NIT than a blocking issue so i'd approve the PR :)

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java
Comment (Steffengreiner) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/AppRoutes.java:11]: Also more of a NIT but: 
The purpose of an enum is to make sure only the defined values are to be used, which makes sense in this context since only these values will provide sensible routing. 
Additionally by combining the Enum Constant with a String value similiar to [previous implentations](https://github.com/qbicsoftware/data-model-lib/blob/5ddc9812bb2445d3b22f1bbf985fc90e40f5fd18/src/main/groovy/life/qbic/datamodel/dtos/business/AcademicTitle.groovy) you'd still get the same result.

**Review by Steffengreiner — APPROVED** [review]
I still have some NITS, but agree that they should not block this PR 👍

**Review by Steffengreiner — APPROVED** [review]
LGTM 👍

**Pull request merged**
